src/data-type.js

```diff
@@ -1305,10 +1305,30 @@ const TYPE = module.exports.TYPE = {
       if (value instanceof Date) {
         return value;
       }
-      value = Date.parse(value);
-      if (isNaN(value)) {
+      var dateValue = Date.parse(value);
+      if (!isNaN(dateValue)) {
```

Comment:
I think you can replace with a simpler version that does not require custom parsing:

```javascript
if (!isNaN(dateValue)) {
  return dateValue;
}
dateValue = Date.parse("2000-01-01 " + value);
if (isNaN(dateValue)) {
  return new TypeError('Invalid date or timespan.');
}

return dateValue;
```